### PR TITLE
kubeadm: remove the ClusterStatus object from v1beta3

### PR DESCRIPTION
**Pull request by neolit123**
#### What this PR does / why we need it:

- Remove the object form v1beta3 and internal version
- Deprecate a couple of phases that were specifically designed / named to
modify the ClusterStatus object
- Adapt logic around annotation vs ClusterStatus retrieval
- Update unit tests
- Run generators

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubeadm/issues/2290
xref https://github.com/kubernetes/kubeadm/issues/1796
fixes https://github.com/kubernetes/enhancements/issues/2506

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: remove the ClusterStatus API from v1beta3 and its management in the kube-system/kubeadm-config ConfigMap. This method of keeping track of what API endpoints exists in the cluster was replaced (in a prior release) by a method to annotate the etcd Pods that kubeadm creates in "stacked etcd" clusters. The following CLI sub-phases are deprecated and are now a NO-OP: for " kubeadm join": "control-plane-join/update-status", for "kubeadm reset": "update-cluster-status". Unless you are using these phases explicitly, you should not be affected.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/2506-Remove-ClusterStatus-from-kubeadm-config
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/970-kubeadm-config
```

### CODE REVIEW

File: cmd/kubeadm/test/resources/cluster_status.go [hunk 1]
```diff
@@ -15,27 +15,3 @@ limitations under the License.
 */
 
 package resources
```
Comment (fabriziopandini) [inline cmd/kubeadm/test/resources/cluster_status.go:17]: Should we drop this file entirely?

**Review by fabriziopandini — COMMENTED** [review]
This is a nice cleanup!
Only a small nit

File: cmd/kubeadm/test/resources/cluster_status.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/test/resources/cluster_status.go]: updated to remove the file.

**Comment by fabriziopandini** [comment]
Great to see the work on v1beta3 moving so fast!
/lgtm

**Pull request merged**